torch/nested/_internal/nested_tensor.py

```diff
@@ -51,20 +61,29 @@ def __new__(cls, values, offsets, **kwargs):
         r._values = values.detach() if values.requires_grad else values
         return r
 
-    def __init__(self, values, offsets, **kwargs):
+    def __init__(self, values, offsets, *, sym_size=None, sym_stride=None, **kwargs):
         super().__init__()
         # Only support jagged for now.
         assert offsets is not None
         assert offsets.ndim == 1
         assert not isinstance(values, NestedTensor)
         assert values.ndim == 2
 
-        # In a later PR, we'll need to accept an additional size argument
-        # to handle dynamic shapes.
-        ragged_dim = get_tensor_id(offsets)
-        D = values.shape[1]
-        B = offsets.shape[0] - 1
-        self._size = (B, ragged_dim, D)
+        if sym_size is not None:
+            assert sym_stride is not None
+            # sym_size and sym_stride are passed during tracing (1) when we
+            # initially fakify the nested tensor, and (2) when we rewrap as we
+            # perform operations on fake nested tensors.
+            self._size = sym_size
+            self._strides = sym_stride
+            self.ragged_size = self._size[1]  # type: ignore[assignment]
+        else:
+            self.ragged_size = get_tensor_id(offsets, 1)
+            D = values.shape[1]
+            B = offsets.shape[0] - 1
+            # TODO: factor out and generalize the stride computing logic
+            self._size = (B, self.ragged_size, D)
+            self._strides = (self.ragged_size * D, D, 1)
```

Comment:
What's going on with the conditional here?

Reply:
yeah I'm unclear on this too. Why do we need a constructor branch that directly accepts `sym_size` / `sym_stride`? Given the same `offsets`, we should produce the same `_size`, `_strides`, `ragged_size`, no?

Reply:
The `else` path here is for during eager, where I want to be able to pass in a offsets tensors and have the  ragged_size  be produced automatically with `get_tensor_id`

During tracing, that is not enough because `get_tensor_id` would return us the non-symbolic size.

I can see an alternative where we have a map somewhere that remembers what offsets map to what SymInts, but I don't have a preference.

Reply:
Actually, I found a better way to do this.